subsys/emul/emul.c

```diff
@@ -46,12 +45,49 @@ int emul_init_for_bus_from_list(const struct device *dev,
 
 		__ASSERT(emul, "Cannot find emulator for '%s'", elp->label);
 
+		switch (emul->bus_type) {
+		case EMUL_BUS_TYPE_I2C:
+			emul->bus.i2c->target = emul;
+			break;
+		case EMUL_BUS_TYPE_ESPI:
+			emul->bus.espi->target = emul;
+			break;
+		case EMUL_BUS_TYPE_SPI:
+			emul->bus.spi->target = emul;
+			break;
+		}
 		int rc = emul->init(emul, dev);
 
 		if (rc != 0) {
-			LOG_WRN("Init %s emulator failed: %d\n",
+			LOG_WRN("Init %s emulator failed: %d",
 				 elp->label, rc);
 		}
+
+		switch (emul->bus_type) {
+		case EMUL_BUS_TYPE_I2C:
+#ifdef CONFIG_I2C_EMUL
```

Comment:
This `#ifdef` should be above the case, or alternatively you should move the `#endif` up so that the `break;` is still there otherwise you'll fall through

Reply:
Oops! Good-catch! Good catch.